core/common/src/Clock.kt

```diff
@@ -15,11 +15,26 @@ import kotlin.time.*
 public interface Clock {
     /**
      * Returns the [Instant] corresponding to the current time, according to this clock.
+     *
+     * It is not guaranteed that calling [now] later will return a larger [Instant].
+     * In particular, for [System], violations of this are completely expected and must be taken into account.
+     * See the documentation of [System] for details.
      */
     public fun now(): Instant
 
     /**
-     * The [Clock] instance that queries the operating system as its source of knowledge of time.
+     * The [Clock] instance that queries the operating system as its source of time knowledge.
+     *
+     * Successive calls to [now] will not necessarily return increasing [Instant] values, and when they do,
+     * these increases will not necessarily correspond to the elapsed time.
+     *
+     * For example, when using [Clock.System], the following could happen:
+     * - [now] returns `2023-01-02T22:35:01Z`;
+     * - The system queries the Internet and recognizes that its clock needs adjusting;
+     * - [now] returns `2023-01-02T22:32:05Z`.
+     *
+     * When predictable intervals between successive measurements are needed, consider using
+     * [TimeSource.Monotonic].
      */
     public object System : Clock {
         override fun now(): Instant = @Suppress("DEPRECATION_ERROR") Instant.now()
```

Comment:
What are some examples of what we could do?

Reply:
We can remove it, write a `@suppress` to avoid [this](https://kotlinlang.org/api/kotlinx-datetime/kotlinx-datetime/kotlinx.datetime/-clock/-companion/) or add a brief explanation "here's the companion to write your extensions on it. By the way, do not use its identity in anyway, it's purely for namespacing"